java/vector/src/main/codegen/templates/UnionFixedSizeListWriter.java

```diff
@@ -169,6 +169,29 @@ public StructWriter struct(String name) {
     return structWriter;
   }
 
+  @Override
+  public MapWriter map() {
+    return writer;
+  }
+
+  @Override
+  public MapWriter map(String name) {
+    MapWriter mapWriter = writer.map(name);
+    return mapWriter;
+  }
+
+  @Override
+  public MapWriter map(boolean keysSorted) {
+    writer.map(keysSorted);
```

Comment:
should be this ?

```Java
MapWriter mapWriter = writer.map(keysSorted);
return mapWriter;
```

Reply:
I had originally thought this too, but making this change will cause the new unit tests I added for MapWriter to break (it won't correctly write a list of maps).

I believe it is because of how List and Map vectors are implemented. Map itself does not store any data, it is a subclass of List, delegating storage to the parent. When we have a list and want to write a map value to get `List<Map<?, ?>>`, we don't need a new writer with a new vector to write into, we actually want to reuse the vector of the current list. If we construct a new writer and return that, we would end up with `List<List<Map<?, ?>>>` when we really want `List<Map<?,?>>`.

My explanation probably isn't 100% clear or correct, but it's how I was able to figure out how to implement MapWriter for this PR.

Reply:
It should work pretty much like `ListWriter` and this isn't how `ListWriter` is done in this class, so something doesn't seem quite right, but I could be wrong. I'll try to look into it and see what's going on.

Reply:
I tried digging into this a little bit more and stepped through what is done for list. I think? we are matching what is done for list, for example [see here](https://github.com/apache/arrow/blob/master/java/vector/src/main/codegen/templates/UnionListWriter.java#L162) where `list()` just returns the current `UnionListWriter`. Since we need `keysSorted` for map, there's an additional `map(boolean)` method, but just like `map()` we return the current writer. Appreciate any more insight you can provide here Bryan.